src/test/java/bio/terra/workspace/integration/WorkspaceIntegrationTest.java

```diff
@@ -158,6 +163,49 @@ public void deleteWorkspaceWithInvalidToken(TestInfo testInfo) throws Exception
     Assertions.assertTrue(deleteWorkspaceResponse.isErrorObject());
   }
 
+  @Test
+  @Tag(TAG_NEEDS_CLEANUP)
+  public void createDataReference(TestInfo testInfo) throws Exception {
+    // This test relies on a persistent snapshot existing in Data Repo, currently in dev.
+    // This snapshot was created using our dev service account, which is a steward in dev Data Repo.
+    // First, I created a dataset "wm_integration_test_dataset" using TDR's
+    // snapshot-test-dataset.json. Then, I created the snapshot "workspace_integration_test_snapshot"
+    // using the "byFullView" mode. Finally, I added the integration test user as a reader of this snapshot.
+    // These steps should only need to be repeated if the dev DataRepo data is deleted, or to support
+    // this test in other DataRepo environments.
+    UUID workspaceId = UUID.randomUUID();
+    testToWorkspaceIdsMap.put(testInfo.getDisplayName(), Collections.singletonList(workspaceId));
+
+    createDefaultWorkspace(workspaceId);
+
+    String userEmail = testConfig.getServiceAccountEmail();
+    String path = testConfig.getWsmWorkspacesBaseUrl() + "/" + workspaceId + "/datareferences";
+
+    DataRepoSnapshot snapshotReference =
+        new DataRepoSnapshot()
+            .snapshot("97b5559a-2f8f-4df3-89ae-5a249173ee0c")
+            .instanceName("terra-dev");
```

Comment:
This can be moved into the properties file https://github.com/DataBiosphere/terra-workspace-manager/blob/dev/src/test/resources/application-integration-test.properties

 and consumed/configured here https://github.com/DataBiosphere/terra-workspace-manager/blob/dev/src/test/java/bio/terra/workspace/integration/common/configuration/TestConfiguration.java

Reply:
Good point, that's a pretty clean way of handling the environment-specific bits. I'll move the instance name there now. The snapshot name would also go there if we use the persistent dev snapshot, but not if we create a dataset with each test run.